sa/database.go

```diff
@@ -122,7 +86,21 @@ func adjustMySQLConfig(conf *mysql.Config) *mysql.Config {
 	// had a different type than what is in the schema, strings being
 	// truncated, writing null to a NOT NULL column, and so on. See
 	// <https://dev.mysql.com/doc/refman/5.0/en/sql-mode.html#sql-mode-strict>.
-	conf.Strict = true
+	conf.Params = make(map[string]string)
+	conf.Params["sql_mode"] = "STRICT_ALL_TABLES"
+
+	// If a read timeout is set, we set max_statement_time to 95% of that, and
+	// long_query_time to 80% of that. That way we get logs of queries that are
+	// close to timing out but not yet doing so, and our queries get stopped by
+	// max_statement_time before timing out the read. This generates clearer
+	// errors, and avoids unnecessary reconnects.
+	if conf.ReadTimeout != 0 {
+		// In MariaDB, max_statement_time and long_query_time are both seconds.
+		// Note: in MySQL (which we don't use), max_statement_time is millis.
+		readTimeout := conf.ReadTimeout.Seconds()
+		conf.Params["max_statement_time"] = fmt.Sprintf("%g", readTimeout*0.95)
```

Comment:
@jsha: random aside, we were prviously using it but do we really want to be using `%g` here? Documentation for it says `%e for large exponents, %f otherwise` where `%e` is `scientific notation, e.g. -1.234456e+78`. Can MariaDB really understand scientific notation or should we just be using `%f` consistently?